--- src/bootstrap/lib.rs.orig	Sat Feb 25 10:19:47 2017
+++ src/bootstrap/lib.rs	Sat Feb 25 10:19:57 2017
@@ -818,7 +818,6 @@
         // gcc-rs because the build scripts will determine that for themselves.
         let mut base = self.cc[target].0.args().iter()
                            .map(|s| s.to_string_lossy().into_owned())
-                           .filter(|s| !s.starts_with("-O") && !s.starts_with("/O"))
                            .collect::<Vec<_>>();
 
         // If we're compiling on OSX then we add a few unconditional flags
